modules/packages/LinearAlgebra.chpl

```diff
@@ -2490,6 +2490,270 @@ proc cosm(A: []) throws {
   }
 }
 
+// Gets sums along specified axes. Method for Rectangular matrix.
+proc sum(A: [?D], param axis=0) where !isCSDom(D) {
+  if axis == 0 then
+    // axis=0 sums along column
+    return forall j in D.dim(1-axis) do (+ reduce abs(A[D.dim(axis), j..j]));
+  else if axis == 1 then
+    // axis=1 sums along row
+    return forall i in D.dim(1-axis) do (+ reduce abs(A[i..i, D.dim(axis)]));
+}
+
+// Gets sums along specified axes. Method for CSR matrix.
+proc sum(A: [?ADom], param axis=0) where isCSDom(ADom) {
+  var (n,t) = ADom.shape;
+  var sps: sparse subdomain(ADom.parentDom);
+  sps += ADom;
+  if axis == 0 {
+    // axis=0 sums along column
+    var b: [0..<t] A.eltType;
+    forall (i,j) in sps {
+      b[j] += A[i,j];
```

Comment:
Here and below, is this meant to be `abs(A[i,j])`?

Reply:
Yes! Thanks for that. I had it right in absSum method for rectangular matrix.